File: HRManager.Frontend/src/app/interfaces/perfilDto.ts
import { CertificacaoDto } from "./certificacaoDto";
import { HabilitacaoDto } from "./habilitacaoDto";

export interface PerfilDto {
  colaboradorId: number;
  nomeCompleto: string;
  cargo: string;
  email: string;
  morada?: string;
  iban?: string;
  habilitacoes: HabilitacaoDto[];
  certificacoes: CertificacaoDto[];
}

File: HRManager.Backend/HRManager.WebAPI/DTOs/PerfilDto.cs
namespace HRManager.WebAPI.DTOs
{
    public class PerfilDto
    {
        // Dados Básicos
        public Guid ColaboradorId { get; set; }
        public string NomeCompleto { get; set; } = string.Empty;
        public string Cargo { get; set; }=string.Empty;
        public string Email { get; set; }=string.Empty;
        // *** ADICIONE ESTAS LINHAS ***
        public string? Morada { get; set; }
        public string? IBAN { get; set; }
        // ******************************

        // Listas Curriculares
        public List<HabilitacaoDto>? Habilitacoes { get; set; }
        public List<CertificacaoDto>? Certificacoes { get; set; }
    }
}
